Slim/Menu/BrowseLibrary/Releases.pm

```diff
@@ -118,12 +119,12 @@ sub _releases {
 
 			if (!$albumArtist) {
 				$addToMainReleases->();
-				next;
+				next unless $userDefinedRoles;
 			}
 		}
 
 		# Roles on other releases
-		foreach my $role ( grep { $_ ne 'ALBUMARTIST' } split(',', $_->{role_ids} || '') ) {
+		foreach my $role ( grep { $_ ne 'ALBUMARTIST' && $_ ne 'ARTIST' } split(',', $_->{role_ids} || '') ) {
```

Comment:
The addition of `ARTIST` here is to not process what we've already processed above, but where we no longer skip to the next track to cover other roles?

Wouldn't this lead to a release potentially be shown multiple times for a given contributor, for their various roles?

Reply:
Yes, intentionally in the case of user-defined roles (similar to COMPOSER/CONDUCTOR in the case of compilations), because by definition they're important to the user.

But you're right, it would also lead to releases being duplicated for other default roles. I'll work on a fix.

Reply:
Change is pushed. Please check the transformation of the `unless` to an `if (...) ;` in the compilation block.